app/src/main/java/com/alphawallet/app/ui/WalletConnectActivity.java

```diff
@@ -102,6 +106,16 @@
 
     private ActionSheetDialog confirmationDialog;
     private ActionSheetDialog walletConnectDialog;
+    private ActionSheetDialog switchChainDialog;
+    private AddEthereumChainPrompt addEthereumChainPrompt;
+    private long switchChainDialogCallbackId = 1;
```

Comment:
Use a single ActionSheetDialog member - in this case use 'walletConnectDialog'. This way we can ensure there's not an existing dialog when it's called.

Reply:
Done.